cpp/src/quantiles/quantiles.hpp

```diff
@@ -80,5 +87,99 @@ void midpoint(int64_t& result, int64_t lhs, int64_t rhs)
 }
 
 } // end of namespace interpolate
+
+namespace detail {
+
+struct QuantileIndex {
+    gdf_size_type lower_bound;
+    gdf_size_type upper_bound;
+    gdf_size_type nearest;
+    double fraction;
+
+    CUDA_HOST_DEVICE_CALLABLE
+    QuantileIndex(gdf_size_type length, double quant)
+    {
+        // clamp quant value.
+        // Todo: use std::clamp if c++17 is supported.
+        quant = std::min(std::max(quant, 0.0), 1.0);
+
+        // since gdf_size_type is int32_t, there is no underflow/overflow
+        double val = quant*(length -1);
+        lower_bound = std::floor(val);
+        upper_bound = static_cast<size_t>(std::ceil(val));
+        nearest = static_cast<size_t>(std::nearbyint(val));
+        fraction = val - lower_bound;
+    }
+};
+
+template<typename T>
+void singleMemcpy(T& res, T const* input)
```

Comment:
Why do we need a trivial wrapper around cudaMemcpy? (and the name is unclear and doesn't follow our `naming_standards`)

Reply:
We don't. This is all from the old quantiles code that i just moved around.